system/jlib/jflz.cpp

```diff
@@ -640,16 +643,29 @@ class jlib_decl CFastLZCompressor : public CInterface, public ICompressor
         size32_t toflush = (inlenblk==COMMITTED)?inlen:inlenblk;
         if (toflush == 0)
             return;
-        assertex(outlen+sizeof(size32_t)*2+toflush+fastlzSlack(toflush)<=blksz);
+        size32_t outSzRequired = outlen+sizeof(size32_t)*2+toflush+fastlzSlack(toflush);
+        if (!dynamicOutSz)
+            assertex(outSzRequired<=blksz);
+        else
+        {
+            if (outSzRequired>dynamicOutSz)
+            {
```

Comment:
the MemoryBuffer (via ensureCapacity) doubles then grows in chunks and the limit ('dynamicOutSz') is set to the new capacity of the MB, so it won't be increasing the checked limit by the minimum and re-entering this block.